src/wasm/wasm-binary.cpp

```diff
@@ -2897,6 +2878,1370 @@ void WasmBinaryReader::readVars() {
+        tags.push_back(getTagName(getU32LEB()));
+        labels.push_back(getU32LEB());
+      }
+      return builder.makeResume(type, tags, labels);
+    }
+    case BinaryConsts::Suspend:
+      return builder.makeSuspend(getTagName(getU32LEB()));
+
+#define BINARY_INT(code)                                                       \
+  case BinaryConsts::I32##code:                                                \
+    return builder.makeBinary(code##Int32);                                    \
+  case BinaryConsts::I64##code:                                                \
+    return builder.makeBinary(code##Int64);
+#define BINARY_FLOAT(code)                                                     \
+  case BinaryConsts::F32##code:                                                \
+    return builder.makeBinary(code##Float32);                                  \
+  case BinaryConsts::F64##code:                                                \
+    return builder.makeBinary(code##Float64);
+#define BINARY_NUM(code)                                                       \
+  BINARY_INT(code)                                                             \
+  BINARY_FLOAT(code)
+
+      BINARY_NUM(Add);
+      BINARY_NUM(Sub);
+      BINARY_NUM(Mul);
+      BINARY_INT(DivS);
+      BINARY_INT(DivU);
+      BINARY_INT(RemS);
+      BINARY_INT(RemU);
+      BINARY_INT(And);
+      BINARY_INT(Or);
+      BINARY_INT(Xor);
+      BINARY_INT(Shl);
+      BINARY_INT(ShrU);
+      BINARY_INT(ShrS);
+      BINARY_INT(RotL);
+      BINARY_INT(RotR);
+      BINARY_FLOAT(Div);
+      BINARY_FLOAT(CopySign);
+      BINARY_FLOAT(Min);
+      BINARY_FLOAT(Max);
+      BINARY_NUM(Eq);
+      BINARY_NUM(Ne);
+      BINARY_INT(LtS);
+      BINARY_INT(LtU);
+      BINARY_INT(LeS);
+      BINARY_INT(LeU);
+      BINARY_INT(GtS);
+      BINARY_INT(GtU);
+      BINARY_INT(GeS);
+      BINARY_INT(GeU);
+      BINARY_FLOAT(Lt);
+      BINARY_FLOAT(Le);
+      BINARY_FLOAT(Gt);
+      BINARY_FLOAT(Ge);
+
+#define UNARY_INT(code)                                                        \
+  case BinaryConsts::I32##code:                                                \
+    return builder.makeUnary(code##Int32);                                     \
+  case BinaryConsts::I64##code:                                                \
+    return builder.makeUnary(code##Int64);
+#define UNARY_FLOAT(code)                                                      \
+  case BinaryConsts::F32##code:                                                \
+    return builder.makeUnary(code##Float32);                                   \
+  case BinaryConsts::F64##code:                                                \
+    return builder.makeUnary(code##Float64);
+
+      UNARY_INT(Clz);
+      UNARY_INT(Ctz);
+      UNARY_INT(Popcnt);
+      UNARY_INT(EqZ);
+      UNARY_FLOAT(Neg);
+      UNARY_FLOAT(Abs);
+      UNARY_FLOAT(Ceil);
+      UNARY_FLOAT(Floor);
+    // UNARY_FLOAT(NearestInt);
```

Comment:
?

Reply:
This can't be handled by the macro because the binary const values are `F32NearestInt` but the operation names are `NearestFloat32` 🫣 

This seems silly enough that it's worth fixing in a preliminary PR.

Reply:
Sgtm, could be a followup PR too.

Reply:
Done as a follow-up in https://github.com/WebAssembly/binaryen/pull/7089